src/lightstep-datasource/datasource.js

```diff
@@ -126,31 +136,116 @@ export class LightStepDatasource {
     return this.q.when({});
   }
 
-  metricFindQuery() {
+  metricFindQuery(query) {
+    const interpolated = this.templateSrv.replace(query, null, 'regex');
+
+    let queryMapper = this.defaultMapper();
+    if (interpolated) {
+      queryMapper = this.parseQuery(interpolated);
+    }
+
     return this.doRequest({
-      url: `${this.url}/public/v0.1/${this.organizationName}/projects/${this.projectName}/searches`,
+      url: `${this.url}/public/${version}/${this.organizationName}/projects/${this.projectName}/streams`,
       method: 'GET',
     }).then(response => {
-      const searches = response.data.data;
-      return _.flatMap(searches, search => {
-        const attributes = search["attributes"];
+      const streams = response.data.data;  
+      return _.flatMap(streams, stream => {
+        const attributes = stream["attributes"];
         const name = attributes["name"];
         const query = attributes["query"];
-        const savedSearchId = search["id"];
+        const streamId = stream["id"];
 
-        // Don't duplicate if the name and query are the same
-        if (name.trim() === query.trim()) {
-          return [ { text: name, value: savedSearchId } ];
-        }
-
-        return [
-          { text: query, value: savedSearchId },
-          { text: name, value: savedSearchId },
-        ];
+        return queryMapper(name, query, streamId);
       });
     });
   }
 
+  defaultMapper() {
+    return (name, query, id) => {
+      // Don't duplicate if the name and query are the same
+      if (name.trim() === query.trim()) {
+        return [ { text: name, value: id } ];
+      }
+
+      return [
+        { text: query, value: id },
+        { text: name, value: id },
+      ];
+    }
+  }
+
+  parseQuery(query) {
+    const matches = query.match(/^(stream_ids|attributes)\(.*/);
+    if (matches && matches.length == 2) {
+      switch (matches[1]) {
```

Comment:
Why is this matches[1] here? I think I'm just confused.

Reply:
`matches[0]` is the full query and `matches[1]` returns the first capturing group. It's used to determine which parsing function will be used later. I started by building a single big regex but it was painful to look at. I could replace it with `.startsWith()` in an `if/else if` structure if preferred.

Reply:
Got it, I think I'm just rusty on my JS regex. No need to change.